src/main/java/edu/emory/cci/aiw/cvrg/eureka/cas/CasProperties.java

```diff
@@ -68,4 +68,37 @@ public boolean isOAuthEnabled() {
 				|| isGitHubAuthEnabled()
 				|| isGlobusAuthEnabled();
 	}
+
+	public boolean isLocalOrLDAPAuthentication(){
+		String authHandlers = getValue("eureka.authHandlers");
+		if(authHandlers == null || authHandlers.isEmpty()){
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * method to identify if the cas-login screen needs to be skipped to go to
+	 * oauth login screen.
+	 * @return
+	 */
+	public boolean isOAuthDirectLogin() {
```

Comment:
This method is so specific to display logic. Either implement the logic in the JSP, or create a separate class that only the JSP creates and uses.